core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -386,6 +388,29 @@ public void nameResolverReturnsEmptyList() {
   }
 
   @Test
+  public void loadBalancerThrowsInHandleResolvedAddresses() {
+    RuntimeException ex = new RuntimeException("simulated");
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(false);
+    ManagedChannel channel = createChannel(nameResolverFactory, NO_INTERCEPTOR);
+    ClientCall<String, Integer> call = channel.newCall(method, CallOptions.DEFAULT);
+    call.start(mockCallListener, new Metadata());
+    assertEquals(1, loadBalancerFactory.balancers.size());
+    LoadBalancer<?> loadBalancer = loadBalancerFactory.balancers.get(0);
+    doThrow(ex).when(loadBalancer).handleResolvedAddresses(
+        Matchers.<List<ResolvedServerInfo>>anyObject(), any(Attributes.class));
+    nameResolverFactory.allResolved();
+    ArgumentCaptor<Status> statusCaptor = ArgumentCaptor.forClass(Status.class);
+    verify(mockCallListener, timeout(1000)).onClose(statusCaptor.capture(), any(Metadata.class));
```

Comment:
Please add some new lines to these test methods.  It makes it easier to visually parse the parts of the test.  Normally tests are split up into SETUP, TEST, and VERIFY sections.  It's good to add a new line between each of those parts.  (and leads to good test organization!)